### ISSUE TITLE
Allow dynamic subscribing to topics, NOT assign

### ISSUE THREAD

**Issue by patricklx** [IBMStreams/streamsx.kafka#103]
Currently its only possible to dynamically assign new topics via `createMessageAddTopicPartition`,
but its not possible to subsribe to new topics and thus we cannot use group management

**Comment by RolefH** [comment]
This involves extending the syntax of JSON and its processing. To the existing syntax
```
{
  "action" : "ADD" or "REMOVE",
  "topicPartitionOffsets" : [
    {
      "topic" : "topic-name",
      "partition" : <partition_number>,
      "offset" : <offset_number>
    },
    ...
  ]
}
```
we would need to add (as an example)
```
{
  "action" : "ADD" or "REMOVE",
  "topics" : [
    {
      "topic" : "topic-name",
    },
    ...
  ]
}
```

**Comment by RolefH** [comment]
Means, that the *partition* element in the JSON would become optional. The *offset* element is always related to a partition. When the *partition* element is missing, the operator would subscribe with fetching from the default position. An *offset* element would be ignored or treated as an error. 

The first ADD action would determine the *subscription mode* = {assign|subscribe} of the operator, which cannot be changed anymore as it determines, which `ConsumerClient` implementation is instantiated. In a consumer group, the user would be responsible to pass the control stream to all consumers of the group.

**Comment by RolefH** [comment]
resolved in [Release 3.0.0](https://github.com/IBMStreams/streamsx.kafka/releases/tag/v3.0.0).
